geode-core/src/main/java/org/apache/geode/internal/cache/FilterRoutingInfo.java

```diff
@@ -386,121 +330,121 @@ public void toData(DataOutput out) throws IOException {
       byte[] myData = StaticSerialization.getThreadLocalByteArray(size);
       hdos = new HeapDataOutputStream(myData);
       hdos.disallowExpansion();
-      if (this.cqs == null) {
+      if (cqs == null) {
         hdos.writeBoolean(false);
       } else {
         hdos.writeBoolean(true);
         InternalDataSerializer.writeArrayLength(cqs.size(), hdos);
-        for (Iterator it = this.cqs.entrySet().iterator(); it.hasNext();) {
-          Map.Entry e = (Map.Entry) it.next();
+        for (final Map.Entry<Long, Integer> longIntegerEntry : cqs.entrySet()) {
           // most cq IDs and all event types are small ints, so we use an optimized
           // write that serializes 7 bits at a time in a compact form
-          InternalDataSerializer.writeUnsignedVL((Long) e.getKey(), hdos);
-          InternalDataSerializer.writeUnsignedVL((Integer) e.getValue(), hdos);
+          InternalDataSerializer.writeUnsignedVL(longIntegerEntry.getKey(), hdos);
+          InternalDataSerializer.writeUnsignedVL(longIntegerEntry.getValue(), hdos);
         }
       }
-      InternalDataSerializer.writeSetOfLongs(this.interestedClients, this.longIDs, hdos);
-      InternalDataSerializer.writeSetOfLongs(this.interestedClientsInv, this.longIDs, hdos);
+      InternalDataSerializer.writeSetOfLongs(interestedClients, longIDs, hdos);
+      InternalDataSerializer.writeSetOfLongs(interestedClientsInv, longIDs, hdos);
```

Comment:
This code block is duplicated in `toDataPre_GFE_8_0_0_0()`. Could it be pulled out into a method, maybe something like "writeCQData()"? If the `toDataPre_GFE_8_0_0_0()` method is shortly to be removed as well though, then there's no need to bother with this.

Reply:
There is a PR pending that removes 8.0.